### PR TITLE
PearAI Submodule Security Updates

### PR DESCRIPTION
**Pull request by cqdev-co**
## Description

Below is a list of security concerns that have been addressed by this pr:
- https://github.com/trypear/pearai-submodule/security/dependabot/35
- https://github.com/trypear/pearai-submodule/security/dependabot/34
- https://github.com/trypear/pearai-submodule/security/dependabot/33
- https://github.com/trypear/pearai-submodule/security/dependabot/32
- https://github.com/trypear/pearai-submodule/security/dependabot/31
- https://github.com/trypear/pearai-submodule/security/dependabot/30
- https://github.com/trypear/pearai-submodule/security/dependabot/29
- https://github.com/trypear/pearai-submodule/security/dependabot/28
- https://github.com/trypear/pearai-submodule/security/dependabot/24
- https://github.com/trypear/pearai-submodule/security/dependabot/23
- https://github.com/trypear/pearai-submodule/security/dependabot/21
- https://github.com/trypear/pearai-submodule/security/dependabot/11
- https://github.com/trypear/pearai-submodule/security/dependabot/20
- https://github.com/trypear/pearai-submodule/security/dependabot/15
- https://github.com/trypear/pearai-submodule/security/dependabot/12
- https://github.com/trypear/pearai-submodule/security/dependabot/9
- https://github.com/trypear/pearai-submodule/security/dependabot/36
- https://github.com/trypear/pearai-submodule/security/code-scanning/5
- https://github.com/trypear/pearai-submodule/security/code-scanning/4
- https://github.com/trypear/pearai-submodule/security/code-scanning/3
- https://github.com/trypear/pearai-submodule/security/code-scanning/6
- https://github.com/trypear/pearai-submodule/security/code-scanning/7
- https://github.com/trypear/pearai-submodule/security/code-scanning/11
- https://github.com/trypear/pearai-submodule/security/code-scanning/10

## Checklist

This checklist is ensuring that pearai-submodule is still running smooth despite these updates.

- [x] Install dependencies runs smoothly 
- [x] Make sure PearAI is working normally
Screenshot 2024-07-09 at 12 39 08 PM
Screenshot 2024-07-09 at 2 00 52 PM

### CODE REVIEW

File: gui/src/components/mainInput/ContextItemsPeek.tsx [hunk 1]
```diff
@@ -43,7 +53,7 @@
 
   function openContextItem(contextItem: ContextItemWithId) {
     if (contextItem.description.startsWith("http")) {
-      window.open(contextItem.description, "_blank");
+      window.open(isSafeUrl(contextItem.description) ? contextItem.description : '#', "_blank");
```
[the remark this answers is not in the document]
Comment (cqdev-co) [reply gui/src/components/mainInput/ContextItemsPeek.tsx:56]: These changes so address these issues so it's weird that they are being flagged

File: gui/src/components/mainInput/ContextItemsPeek.tsx [hunk 2]
```diff
@@ -108,7 +118,7 @@
               return (
                 <a
                   key={idx}
-                  href={contextItem.description}
+                  href={isSafeUrl(contextItem.description) ? contextItem.description : '#'} // Protects against Client-side URL Redirects
```
[the remark this answers is not in the document]
Comment (cqdev-co) [reply gui/src/components/mainInput/ContextItemsPeek.tsx:121]: These changes so address these issues so it's weird that they are being flagged

File: gui/src/components/mainInput/ContextItemsPeek.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (nang-dev) [reply gui/src/components/mainInput/ContextItemsPeek.tsx:56]: lol

**Review by nang-dev — APPROVED** [review]
nice! thanks so much 🔒 🔒

**Pull request merged**